tests/matrix_free/point_evaluation_19.cc

```diff
@@ -0,0 +1,167 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 - 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Check FEPointEvaluation::reinit with precomputed mapping
+// information for a vector valued FE and MappingCartesian.
+// This is a modified version of point_evaluation_13.cc.
+
+#include <deal.II/base/function_lib.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/mapping_cartesian.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/matrix_free/fe_point_evaluation.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+#include <iostream>
+
+#include "../tests.h"
+
+
+
+template <int dim>
+class MyFunction : public Function<dim>
+{
+public:
+  MyFunction()
+    : Function<dim>(dim)
+  {}
+
+  double
+  value(const Point<dim> &p, const unsigned int component) const override
+  {
+    return component + p[component];
+  }
+};
+
+
+
+template <int dim>
+void
+test()
+{
+  using namespace dealii;
+  Triangulation<dim> tria;
+
+  GridGenerator::subdivided_hyper_cube(tria, 2, 0, 1);
+
+  MappingCartesian<dim> mapping;
+  deallog << "Cartesian linear mapping" << std::endl;
+
+  std::vector<Point<dim>> unit_points;
+  for (unsigned int i = 0; i < 13; ++i)
+    {
+      Point<dim> p;
+      for (unsigned int d = 0; d < dim; ++d)
+        p[d] = static_cast<double>(i) / 17. + 0.015625 * d;
+      unit_points.push_back(p);
+    }
+
+  std::vector<Point<dim>> unit_points2;
+  for (unsigned int i = 0; i < 15; ++i)
+    {
+      Point<dim> p;
+      for (unsigned int d = 0; d < dim; ++d)
+        p[d] = static_cast<double>(i) / 17. + 0.015625 * d;
+      unit_points2.push_back(p);
+    }
+
+  FESystem<dim>   fe(FE_Q<dim>(2), dim);
+  DoFHandler<dim> dof_handler(tria);
+  dof_handler.distribute_dofs(fe);
+  Vector<double> vector(dof_handler.n_dofs());
+
+  // mapping info object for precomputed mapping
+  NonMatching::MappingInfo<dim, dim> mapping_info(mapping,
+                                                  update_values |
+                                                    update_gradients);
+  // First evaluator to use precomputed mapping
+  FEPointEvaluation<dim, dim> evaluator1(mapping_info, fe);
+
+  // Second evaluator to use precomputed mapping
+  FEPointEvaluation<dim, dim> evaluator2(mapping_info, fe);
```

Comment:
This imitates the usage in https://github.com/geodynamics/aspect/blob/a2d8beb3511d6548be4547fb203565719a04e91f/source/particle/world.cc#L618 that's why I kept it. This test is focused on a different number of unit points. Do you think we should skip this since this is checked in other tests?

Reply:
I think we can keep the `mapping_info.reinit()` part, but I would not use two evaluators for the sole purpose of sharing the data. We have other tests for these scenarios.